lib/state/network.dart

```diff
@@ -26,20 +27,20 @@ class Network {
   const Network({required this.name, required this.walletProxyConfig});
 }
 
-class ActiveNetworkState {
-  /// Currently active network.
+class SelectedNetworkState {
+  /// Services corresponding to the currently selected network (as defined in [Config.availableNetworks]).
   ///
-  /// The network is guaranteed to be one of the "enabled" networks (as defined in [Config.availableNetworks]).
-  Network active;
+  /// The network is guaranteed to be one of the entries in [ServiceRepository.activeNetworks].
+  final NetworkServices services;
 
-  ActiveNetworkState(this.active);
+  const SelectedNetworkState(this.services);
 }
 
-/// State component acting as the source of truth for what network is currently active in the app.
-class ActiveNetwork extends Cubit<ActiveNetworkState> {
-  ActiveNetwork(Network active) : super(ActiveNetworkState(active));
+/// State component acting as the source of truth for what network is currently selected in the app.
+class SelectedNetwork extends Cubit<SelectedNetworkState> {
+  SelectedNetwork(NetworkServices services) : super(SelectedNetworkState(services));
 
-  void setActive(Network n) {
-    emit(ActiveNetworkState(n));
+  void select(NetworkServices networkServices) {
```

Comment:
Currently anywhere in the application the network can be changed and the requirements written in method documentation in this class does not necessarily hold try.

I think it would be more clean if the select took a `Network` as input and then from `[ServiceRepository.activeNetworks]` either created or returned the `NetworkServices`. Hence there would exist some kind of `SelectedNetworkManager`.

By this any Widget only needs to known about a `Network`.

Reply:
That may be an option but it would require the bloc to know about `ServiceRepository` in order to activate or lookup the network services. This is discouraged by the bloc library, which suggests that you wire blocs together via listeners or similar. Which is what you can do here by looking up the network or activate it async and then pass the result to this method. And yes that could/should probably be done by some kind of network manager.

I'll keep it like this, but you're welcome to try out your idea.

I'm unsure what you mean about the class docs not being accurate? This bloc defines what network is selected no matter where it's selected from.